src/main/kotlin/org/rust/lang/core/resolve/ref/RsPathReferenceImpl.kt

```diff
@@ -81,6 +85,14 @@ fun resolvePath(path: RsPath, lookup: ImplLookup = ImplLookup.relativeTo(path)):
         }
     }
 
+    val lifetimeArguments: List<Re>? =
+        path.typeArgumentList
+            ?.lifetimeList
+            ?.map { lifetime ->
+                val resolved = lifetime.reference.resolve() as? RsLifetimeParameter
+                resolved?.let { ReEarlyBound.named(it) } ?: ReUnknown
```

Comment:
You can write something like this
```kotlin
val resolved = lifetime.reference.resolve() as? RsLifetimeParameter ?: return@map ReUnknown
ReEarlyBound.named(it)
```
to get rid of `let` here - it ususally makes code readability worse

Reply:
Fixed

Reply:
I am a little late here, but maybe we should rewrite it like this?
```kotlin
       map { lifetime ->
            val resolved = lifetime.reference.resolve()
            if (resolved is RsLifetimeParameter)
                ReEarlyBound.named(resolved)
            else
                ReUnknown
        }
```

It is much more clean about what kind of decision we are doing in this closure, IMO

Reply:
I prefer to use [Google's convention](https://android.github.io/kotlin-guides/style.html#expressions) about `if`/`else` expressions:
> An `if`/`else` conditional that is used as an expression may omit braces only if the entire expression fits on one line.